app/models/components/course/model_component_host.rb

```diff
@@ -26,6 +26,18 @@ def after_course_create(_course)
     end
   end
 
+  module Duplicable
+    extend ActiveSupport::Concern
+
+    def initialize_duplicate(_duplicator, _other)
+      fail 'Implement your own initialize_duplicate method for this component.'
+    end
+
+    def duplicator_display_string
+      fail 'Return the string to display on the duplication UI page.'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.